internal/pkg/metrics/kyma.go

```diff
@@ -142,3 +147,72 @@ func calcStateValue(state, newState shared.State) float64 {
 func (k *KymaMetrics) RecordRequeueReason(kymaRequeueReason KymaRequeueReason, requeueType RequeueType) {
 	k.requeueReasonCounter.WithLabelValues(string(kymaRequeueReason), string(requeueType)).Inc()
 }
+
+func (k *KymaMetrics) CleanupNonExistingKymaCrsMetrics(ctx context.Context, kcpClient client.Client) error {
+	currentLifecycleManagerMetrics, err := FetchLifecycleManagerMetrics()
+	if err != nil {
+		return err
+	}
+
+	if len(currentLifecycleManagerMetrics) == 0 {
+		return nil
+	}
+
+	kymaCrsList := &v1beta2.KymaList{}
+	err = kcpClient.List(ctx, kymaCrsList)
+	if err != nil {
+		return fmt.Errorf("failed to fetch Kyma CRs, %w", err)
+	}
+	kymaNames := getKymaNames(kymaCrsList)
+	for _, m := range currentLifecycleManagerMetrics {
+		currentKymaName := getKymaNameFromLabels(m)
+		if currentKymaName == "" {
+			continue
+		}
+		if _, exists := kymaNames[currentKymaName]; !exists {
+			logs.FromContext(ctx).Info(fmt.Sprintf("Deleting a metric for non-existing Kyma: %s", currentKymaName))
+			k.KymaStateGauge.DeletePartialMatch(prometheus.Labels{
+				KymaNameLabel: currentKymaName,
+			})
```

Comment:
I just noticed that there is a `CleanupMetrics` function which seems to do the same thing + also deleting `lifecycle_mgr_module_state` for respective kyma name. Can this be re-used or do we not want to delete the module-related ones?

https://github.com/kyma-project/lifecycle-manager/blob/main/internal/pkg/metrics/kyma.go#L91-L98

Reply:
I didn't use it because of the module-related metrics. Right now only the lifecycle manager are the ones that remain, that's why I didn't cleanup the modules metrics